tests/hawtio-test-suite/src/main/java/io/hawt/tests/features/stepdefinitions/security/ThrottlingStepDefs.java

```diff
@@ -0,0 +1,68 @@
+package io.hawt.tests.features.stepdefinitions.security;
+
+import com.codeborne.selenide.Selenide;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.hawt.tests.features.config.TestConfiguration;
+import io.hawt.tests.features.pageobjects.pages.ConnectPage;
+import io.hawt.tests.features.pageobjects.pages.LoginPage;
+import io.hawt.tests.features.setup.LoginLogout;
+import org.openqa.selenium.By;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.codeborne.selenide.Condition.text;
+import static com.codeborne.selenide.Selenide.$;
+
+public class ThrottlingStepDefs {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ThrottlingStepDefs.class);
+    private final LoginPage loginPage = new LoginPage();
+    private final ConnectPage connectPage = new ConnectPage();
+    private static final By WARNING = By.cssSelector(".pf-v5-c-helper-text__item-text");
+    private static final By CONNECT_PAGE_ALERT = By.cssSelector(".pf-v5-c-alert__title");
+
+//    private int numWindows() {
+//        return WebDriverRunner.getWebDriver().getWindowHandles().size();
+//    }
+
+    @Given("User is on Login page")
+    public void userIsOnLoginPage() {
+        if (TestConfiguration.getConnectUrl() != null) {
+            //LoginLogout.logout();
+            Selenide.closeWindow();
+            Selenide.switchTo().window(0);
+            connectPage.connectTo("test-connection");
+            //loginPage.loginPageIsOpened();
+        } else {
+            LoginLogout.logout();
+            loginPage.loginPageIsOpened();
+        }
+    }
+
+    @When("the user attempts to log in with incorrect credentials {int} times")
+    public void theUserAttemptsToLogInWithIncorrectCredentialsTimes(int attempts) {
+        if (TestConfiguration.getConnectUrl() != null) {
+            for (int i = 0; i < attempts; i++) {
+                connectPage.login("username", "invalid");
```

Comment:
You could create a lambda based on the `TestConfiguration.getConnectUrl() != null`
`BiConsumer<String, String> login = TestConfiguration.getConnectUrl() != null ? connectPage::login : loginPage::login`
and just run `login.accept("username", "invalid")` in the loop :nerd_face: